src/extended_json.ts

```diff
@@ -216,16 +217,16 @@ function serializeValue(value: any, options: EJSONSerializeOptions): any {
   }
 
   if (typeof value === 'number' && (!options.relaxed || !isFinite(value))) {
-    // it's an integer
-    if (Math.floor(value) === value) {
-      const int32Range = value >= BSON_INT32_MIN && value <= BSON_INT32_MAX,
-        int64Range = value >= BSON_INT64_MIN && value <= BSON_INT64_MAX;
+    if (Number.isInteger(value) && !Object.is(value, -0)) {
+      const int32Range = value >= BSON_INT32_MIN && value <= BSON_INT32_MAX;
+      const int64Range = value >= BSON_INT64_MIN && value <= BSON_INT64_MAX;
```

Comment:
optional
```suggestion
      const isInInt32Range = value >= BSON_INT32_MIN && value <= BSON_INT32_MAX;
      const isInInt64Range = value >= BSON_INT64_MIN && value <= BSON_INT64_MAX;
```

Reply:
(and corresponding usages ofc)

Reply:
made the two blocks that do this logic look identical now